File: frontend/src/types/assessment.ts
export type Difficulty = 'easy' | 'medium' | 'hard'

export interface AssessmentQuestion {
  id: string
  question_text: string
  options: string[]
  module_id: number
  difficulty: Difficulty
}

export interface AssessmentStartResponse {
  session_id: string
  questions: AssessmentQuestion[]
  source: 'ai' | 'bank'
}

export interface ModuleScoreBreakdown {
  module_id: number
  module_title: string
  correct: number
  total: number
  percentage: number
}

export interface AssessmentSubmitResponse {
  level: StudentLevel
  score: number
  confidence: number
  module_breakdown: ModuleScoreBreakdown[]
  feedback: string
}

export type StudentLevel = 'beginner' | 'intermediate' | 'advanced'

File: backend/app/schemas/assessment.py
"""Schemas for entry assessment (personalization)."""
import uuid
from datetime import datetime
from pydantic import BaseModel, Field


class AssessmentQuestionResponse(BaseModel):
    id: str
    question_text: str
    options: list[str]
    module_id: int
    difficulty: str  # easy | medium | hard


class AssessmentStartResponse(BaseModel):
    session_id: uuid.UUID
    questions: list[AssessmentQuestionResponse]
    source: str  # "ai" | "bank"


class AssessmentSubmitRequest(BaseModel):
    session_id: uuid.UUID
    answers: dict[str, int]  # {qid: selected_index}


class ModuleScoreBreakdown(BaseModel):
    module_id: int
    module_title: str
    correct: int
    total: int
    percentage: float


class AssessmentSubmitResponse(BaseModel):
    level: str
    score: float
    confidence: float
    module_breakdown: list[ModuleScoreBreakdown]
    feedback: str
